aries_cloudagent/wallet/routes.py

```diff
@@ -757,6 +786,90 @@ async def wallet_set_did_endpoint(request: web.BaseRequest):
         return web.json_response({"txn": transaction.serialize()})
 
 
+@docs(tags=["wallet"], summary="Create a EdDSA jws using did keys with a given payload")
+@request_schema(JWSCreateSchema)
+@response_schema(WalletModuleResponseSchema(), description="")
+async def wallet_jwt_sign(request: web.BaseRequest):
+    """
+        Request handler for jws creation using did.
+
+    Args:
+        "headers": { ... },
+        "payload": { ... },
+        "did": "did:example:123",
+        "verificationMethod": "did:example:123#keys-1"
+        with did and verification being mutually exclusive.
+    """
+    context: AdminRequestContext = request["context"]
+    body = await request.json()
+    did = body.get("did")
+    if did is None:
+        did = body.get("verificationMethod").split("#", 1)[0]
+    if did is None:
+        raise web.HTTPBadRequest(reason="did and verificationMethod required.")
+    payload = body["payload"]
+    headers = body.get("headers", {})
+    async with context.session() as session:
+        wallet = session.inject_or(BaseWallet)
+        if not wallet:
+            raise web.HTTPForbidden(reason="No wallet available")
+        try:
+            did_info = await wallet.get_local_did(did)
+            headers["alg"] = "EdDSA"
+            headers["typ"] = "JWT"  # TODO: why not JWS?
+            headers["kid"] = f"{did}#keys-1"  # TODO: fix me
+            headers["crit"] = ["b64"]  # TODO: do we need this
+            # TODO: do we need to b64encode before signing?
+            signature = await wallet.sign_message(
+                f"{headers.encode('utf-8')}.{payload.encode('utf-8')}", did_info.verkey
```

Comment:
Yes, we need to do both a `json.dumps` and a `bytes_to_b64` for both the headers and the payload.